pkg/obfuscator/report_types.go

```diff
@@ -0,0 +1,31 @@
+package obfuscator
+
+type ReplacementReport struct {
+	Replacements []Replacement
+}
+
+type Replacement struct {
+	Canonical    string       `yaml:"canonical,omitempty"`
+	ReplacedWith string       `yaml:"replacedWith,omitempty"`
+	Occurrences  []Occurrence `yaml:"occurrences,omitempty"`
+}
+
+type Occurrence struct {
+	Original string `yaml:"original,omitempty"`
+	Count    uint   `yaml:"count,omitempty"`
+}
+
+func (r *Replacement) Increment(original string, count uint) Replacement {
+	if r.Occurrences == nil {
+		r.Occurrences = []Occurrence{}
+	}
+	for i, o := range r.Occurrences {
```

Comment:
this is entirely premature optimization since the whole thing is bottlenecked on regex. why not use a map here?

Reply:
we can turn this into a slice later in the Report method without paying the scan price during runtime

Reply:
1. I'm not concerned about the performance here: Eg for MAC addresses we can have up to 4 elements in the slice (dots, hyphens, upper/lower case)
2. I can change it to a map, but I don't think we'll benefit from it at all.

Reply:
you'd be surprised, but I found a must-gather with more than a million IPs:
https://github.com/openshift/must-gather-clean/pull/50